gcp/appengine/frontend_handlers.py

```diff
@@ -40,7 +43,8 @@
 _PAGE_SIZE = 16
 _PAGE_LOOKAHEAD = 4
 _REQUESTS_PER_MIN = 30
-_WORD_CHARACTERS_OR_DASH = re.compile(r'^[\w-]+$')
+_WORD_CHARACTERS_OR_DASH = re.compile(r'^[\w\-]+$')
+_VALID_FILE_NAME = re.compile(r'^[\w\-_]+(\.\w+)*$')
```

Comment:
Is _ needed in the first `['\w\-_]` ? `\w` already includes underscores right? 

Also, can you add a comment on what you're trying to ensure here? The regex is starting to get a bit  complicated to understand the intention :)

Reply:
Ah you're right, _ is not needed, in fact the regex is not actually needed, but just another layer to stop any invalid names. 

It seems like flask blueprint routes already strip out any / (even url encoded), and we are using `send_from_directory`, so this is actually overkill, so I removed the regex now to simplify things.